pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -493,6 +495,46 @@ func (s *server) search(ctx context.Context, cfg *payload.Search_Config,
 			if num != 0 && len(res.GetResults()) > num {
 				res.Results = res.GetResults()[:num]
 			}
+
+			if errors.Is(ectx.Err(), context.DeadlineExceeded) {
```

Comment:
may I confirm that the deadline exceed error will be returned only when all of the agents timeout.

If one of the agent timeout this error will not be returned?

Reply:
This error occurs when the errgroup timeout occurs. So, if no result is returned from at least one agent, this error will occur.

https://github.com/vdaas/vald/blob/25970cf87669f9ea0d65cf7ecbfcfc7e300d9793/pkg/gateway/lb/handler/grpc/handler.go#L365